addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -100,6 +100,8 @@ def _prepare_invoice_values(self, order, name, amount, so_line):
                 'analytic_account_id': order.analytic_account_id.id or False,
             })],
         }
+        if hasattr(order, 'referrer_id'):
```

Comment:
If you need to call `hasattr`, it is probably not done in the right module

Reply:
Hello,

Thank you for your review,
Sorry I though you're a dev for sale as well, I apologize for pinging you in this
and for `hasattr` I'm calling it since this function is being used by some other module that doesn't contain the `referrer_id`

Reply:
> sometime is not a sale.order object

Wait what? In what case? Even if it is true, use `if order._name == 'sale.order'` then, it is way more explicit

Reply:
Sorry I didn't describe the above comment, what I meant is sometimes the `sale.order` object doesn't have a `referrer_id` as there's a test in runbot that was failed before when I did assume that each time the `order` object has a `referrer_id`
please check this build on runbot for an old PR:
https://runbot.odoo.com/runbot/build/15792952

Reply:
Ok, then I get back to my first comment :)

> If you need to call `hasattr`, it is probably not done in the right module

If you search for `referrer_id = fields` in the code, you will see that it has been added in the module `partner_commission`, depending on `sale_subscription` and therefore on `sale`. So you can simply add the new value in an extension of `_prepare_invoice_values` in that module.
We want to keep the logic of modularity without having stuff related to other modules in unrelated ones.